File: CuttingPolygonsC/src/vertex.cpp
#include "vertex.h"
namespace VertexNamespace
{
Vertex::Vertex()
{
    Id=0;
    _point=Point();
}

Vertex::~Vertex()
{

}

Vertex::Vertex(const int &id, const double &x, const double &y)
{
   Id=id;
   _point=Point(x,y);
}

Vertex::Vertex(const int &id, const Point & point)
{
    Id=id;
    _point = point;
}

Point Vertex::GetPoint() const
{
    return _point;
}



double Vertex::GetX() const
{
    return _point.GetX();
}

double Vertex::GetY()  const
{
    return _point.GetY();
}

Vertex &Vertex::operator =(const Vertex &vertex)
{
    Id=vertex.Id;
    _point=vertex._point;
    return *this;
}

};

File: cuttingPolygonsPython/src/vertex.py
import src.point as point


class Vertex:
    Id = 0

    def __init__(self):
        self.Id = 0
        self._point = point.Point()

    def setVertex(self, id: int, x: float, y: float):  #creo un metodo perchè non posso avere 2 costruttori, invece di un altro init (costruttore)
        self.Id = id
        self._point.setcoordinate(x, y)

    def setPoint(self, id: int, point: point.Point):
        self.Id=id
        self._point.setcoordinate(point.getX(), point.getY()) #assegno il punto al vertice
        return self #ritorna il vertice stesso

    def getPoint(self):
        return self._point

    def getX(self ):
        return self._point.getX()

    def getY(self):
        return self._point.getY()





